datashader/transfer_functions/_cuda_utils.py

```diff
@@ -75,32 +75,31 @@ def masked_clip_2d(data, mask, lower, upper):
 # Behaviour of numba.cuda.atomic.max/min changed in 0.50 so as to behave as per
 # np.nanmax/np.nanmin
 if LooseVersion(numba.__version__) >= LooseVersion("0.51.0"):
-    @cuda.jit
-    def cuda_atomic_nanmin(ary, idx, val)
+    @numba.jit
+    def cuda_atomic_nanmin(ary, idx, val):
         return cuda.atomic.nanmin(ary, idx, val)
-    @cuda.jit
-    def cuda_atomic_nanmax(ary, idx, val)
+    @numba.jit
+    def cuda_atomic_nanmax(ary, idx, val):
         return cuda.atomic.nanmax(ary, idx, val)
-elif LooseVersion(numba.__version__) <= LooseVersion("0.49.0"):
-    @cuda.jit
-    def cuda_atomic_nanmin(ary, idx, val)
+elif LooseVersion(numba.__version__) <= LooseVersion("0.49.1"):
+    @numba.jit
+    def cuda_atomic_nanmin(ary, idx, val):
         return cuda.atomic.min(ary, idx, val)
 
-    @cuda.jit
-    def cuda_atomic_nanmax(ary, idx, val)
+    @numba.jit
+    def cuda_atomic_nanmax(ary, idx, val):
         return cuda.atomic.max(ary, idx, val)
 else:
-    # WHAT DO YOU WANT TO DO HERE?
-    pass
+    raise ImportError("Datashader's CUDA support requires numba!=0.50.0")
 
 
 @cuda.jit
 def masked_clip_2d_kernel(data, mask, lower, upper):
     i, j = cuda.grid(2)
     maxi, maxj = data.shape
     if i >= 0 and i < maxi and j >= 0 and j < maxj and not mask[i, j]:
-        cuda_atomic_max(data, (i, j), lower)
-        cuda_atomic_min(data, (i, j), upper)
+        cuda_atomic_nanmax(data, (i, j), lower)
+        cuda_atomic_nanmax(data, (i, j), upper)
```

Comment:
Changing `cuda_atomic_min` to `cuda_atomic_nanmax` looks a little strange here, as the original PR #934 kept `max` and `min`: https://github.com/holoviz/datashader/pull/934/files#diff-d28b7c34f16db5fd95f4c3a0723b9dc9R102-R103 - is this fixing another issue, or is it a copy/paste issue?

Reply:
Seems to be something @jbednar suggested as a change in the original PR. Not quite sure why either. I don't think it makes much of a difference but also doesn't seem necessary.

Reply:
@jbednar suggested changing `cuda_atomic_min` to `cuda_atomic_nanmin` - this changes it to `cuda_atomic_nanmax` - i.e. changing a `min` into a `max, which seems a more significant change.

Reply:
Oh, totally missed that. Good catch!

Reply:
Apologies for the confusion, this was a copy/paste issue, it should be `cuda_atomic_nanmin`.  I have updated the code

Reply:
Thanks for the fix, and thanks @gmarkall for catching it.